server/src/modules/users/application/use-cases/create-user/__tests__/create-user-controller.test.unit.ts

```diff
@@ -1,122 +1,85 @@
-import httpMocks from 'node-mocks-http'
+import { mocks } from '../../../../../../test-utils'
 import { AppError } from '../../../../../../shared/core/app-error'
 import { Result } from '../../../../../../shared/core/result'
-import { DecodedExpressRequest } from '../../../../../../shared/infra/http/routes/decoded-request'
-import { User } from '../../../../domain/entities/user'
 import { UserValueObjectErrors } from '../../../../domain/value-objects/errors'
+import { CreateUserController } from '../create-user-controller'
 import { CreateUserDTO } from '../create-user-dto'
 import { CreateUserErrors } from '../create-user-errors'
 import { CreateUserUseCase } from '../create-user-use-case'
-import { buildController } from '../test-utils/build-controller'
-import { createMockUser } from '../test-utils/create-user'
 
 // TODO: how to show developer these mocks are necessary when building a controller? aka must be synced with buildController()
-jest.mock('../../../../infra/repos/implementations/mikro-user-repo')
 jest.mock('../create-user-use-case')
 
 describe('CreateUserController', () => {
-  let createUserDTO: CreateUserDTO
-  let mockUser: User
-
-  beforeAll(() => {
-    createUserDTO = {
-      email: 'john.doe@uwaterloo.ca',
-      password: 'secret',
-    }
-
-    mockUser = createMockUser()
+  const createUserDTO: CreateUserDTO = {
+    email: 'john.doe@uwaterloo.ca',
+    password: 'secret',
+  }
+  let createUserController: CreateUserController
+  beforeAll(async () => {
+    const createUser = await mocks.mockCreateUser()
+    createUserController = createUser.createUserController
   })
 
   test('When the CreateUserUseCase returns Ok, the CreateUserController returns 200 OK', async () => {
-    const mockRequest = httpMocks.createRequest({
-      body: createUserDTO,
-    }) as DecodedExpressRequest
-    const mockResponse = httpMocks.createResponse()
-    jest.spyOn(CreateUserUseCase.prototype, 'execute').mockResolvedValue(Result.ok(mockUser))
-    const createUserController = buildController()
+    const user = mocks.mockUser(createUserDTO)
```

Comment:
```suggestion
    const user = mocks.createMockUser(createUserDTO)
```

Reply:
I decided on the name because I've seen people use "mock" as an informal verb in the past. `mockUser` _mocks_ a object of type `User` by returning a mock `User`. Thoughts?

Reply:
Ah, it makes sense when I read `mock` as a verb. It's just not that intuitive to me. Feel free to make the final call though.

Reply:
I think I'll stick with `mock` as a verb for the sake of keeping names and lines shorter